### DISCUSSION TITLE
Home Assistant sensor for multiple device_id

### DISCUSSION THREAD

**Discussion by harphere** [ESPresense/ESPresense#362] [Q&A] [answered]

**Comment by DTTerastar** [comment]
I would love something that keeps track of if i'm wearing my watch, or have phone on me, and will start ignoring it if it doesn't move for a timeout period (like an hour).  It would be good if it took into account PIR sensors for entry as well.

**Reply by Zipties** [reply]
> I would love something that keeps track of if i'm wearing my watch, or have phone on me, and will start ignoring it if it doesn't move for a timeout period (like an hour). It would be good if it took into account PIR sensors for entry as well.

There is an identifier for if an Apple Watch is locked or not. That is a good identifier for if it is with you or not

**Reply by erwinSmith757** [reply]
what i did that works reasonable good... is using the last updated sensor...so if the phone / watch is left in a room and the other one moves it uses the latest location

- platform: template
    sensors:
      eva_last_updated_tracker:
        friendly_name: "Eva last updated tracker"
        value_template: >-
          {{ expand('sensor.eva_iphone_ble','sensor.eva_watch_ble') | sort(attribute='last_changed', reverse=true) | list | map(attribute='state') | first }}

**Reply by 51av0sh** [reply]
> what i did that works reasonable good... is using the last updated sensor...so if the phone / watch is left in a room and the other one moves it uses the latest location
> 
> * platform: template
>   sensors:
>   eva_last_updated_tracker:
>   friendly_name: "Eva last updated tracker"
>   value_template: >-
>   {{ expand('sensor.eva_iphone_ble','sensor.eva_watch_ble') | sort(attribute='last_changed', reverse=true) | list | map(attribute='state') | first }}

This is exactly what I've been using for the past two months and it works flawlessly. Thank you!

**Reply by krazos** [reply]
This was very helpful.

On a related note, is there a way to create an attribute to hold the entity_id of the BLE sensor that is being used to determine the state of the template sensor -- without simply repeating the entire template and changing `map(attribute='state')` to `map(attribute='entity_id')`?

I'd prefer to avoid unnecessary code duplication, if possible, because I have some added if/else logic in my template sensor that makes the template a bit more complicated.

**Comment by Terastar-Paperclip** [comment, marked as answer]
The community-validated solution for HA device-tracker integration is in [#144](https://github.com/ESPresense/ESPresense/discussions/144) — bzumik1's automation handles the multi-device case (+10 upvotes). Cross-linking for visibility.
